Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -549,7 +549,7 @@ private static CompletionSuggestion createCompletionSuggestionFromAvailableSugge
       for (String tag : relevanceTags) {
         IncludedSuggestionRelevanceTag relevanceTag = includedSuggestionRelevanceTags.get(tag);
         if (relevanceTag != null) {
-          relevance += includedSuggestionRelevanceTags.get(tag).getRelevanceBoost();
+          relevance = Math.max(relevance, relevanceTag.getRelevanceBoost());
```

Comment:
Hm... "the maximum relevance *boost* is used".
Which means that the boost is *added* to the base relevance.

Reply:
This was my understanding too. It should be `base relevance + (maximum of tag boosts)`.

Reply:
Cool makes sense, updated!

Reply:
@scheglov, is it safe to start with zero assuming that there always be at least one non-negative relevance boost?

@lambdabaa, there are some not related files in the commit :)

Reply:
Thanks for pointing that out. `git commit --amend` not actually the same as `git rebase -i HEAD~2`...

Reply:
> @scheglov, is it safe to start with zero assuming that there always be at least one non-negative relevance boost?

There may be no relevance boosts but then 0 boost is what we want.

Reply:
Yes, there are no negative boosts.